src/object.cr

```diff
@@ -104,10 +104,38 @@ class Object
   # as this will in turn call `to_s(io)` on it.
   abstract def to_s(io : IO)
 
-  # Returns a `String` representation of this object.
+  # Returns a `String` representation of this object suitable
+  # to be embedded inside other expressions, sometimes providing
+  # more information about this object.
   #
-  # Similar to `to_s`, but usually returns more information about
-  # this object.
+  # `#inspect` (and `#inspect(io)`) are the methods used when
+  # you invoke `#to_s` or `#inspect` on an object that holds
+  # other objects and wants to show them. For example when you
+  # invoke `Array#to_s`, `#inspect` will be invoked on each element:
+  #
+  # ```
+  # ary = ["one", "two", "three, etc."]
+  # ary.inspect # => ["one", "two", "three, etc."]
+  # ```
+  #
+  # Note that if Array invoked `#to_s` on each of the elements
+  # above, the output would have been this:
+  #
+  # ```
+  # ary = ["one", "two", "three, etc."]
+  # # If inspect invoked to_s on each element...
```

Comment:
Already stated above and also not stated in the previous example.
```suggestion
```

Reply:
This is if you read the example out of context, to avoid confusion.